presto-main/src/test/java/com/facebook/presto/failureDetector/TestHeartbeatFailureDetector.java

```diff
@@ -88,6 +93,34 @@ public void configure(Binder binder)
         assertTrue(detector.getFailed().isEmpty());
     }
 
+    @Test
```

Comment:
I have a few comments here:
- We don't mark local variables as `final`.
- The default `ObjectMapper` can already serialize `Throwable` type without issues. It fails when we disable introspection/auto-detection in the `ObjectMapper` (see `io.airlift.json.ObjectMapperProvider::get()`), so it's better if you get an `ObjectMapper` from that provider class.
- I think it may be better if we do a round trip to deserialize the serialized value of `Stats` and assert its fields instead of asserting string contents, but that may need changes to `Stats` (have a default constructor so that the json codec can deserialize it back). @electrum What do you think?

So overall I think the test can be rewritten as

```
    @Test
    public void testHeartbeatStatsSerialization()
            throws Exception
    {
        ObjectMapper objectMapper = new ObjectMapperProvider().get();
        Stats stats = new Stats(new URI("http://example.com"));
        String serializedStats = objectMapper.writeValueAsString(stats);
        assertFalse(serializedStats.contains("lastFailureException")); // see my comment above regarding renaming lastFailureException to lastFailureInfo

      // also instead of asserting string contents we may need to deserialize the string back to a Stats instance and do the assertions.

        stats.recordFailure(new SocketTimeoutException("timeout"));
        serializedStats = objectMapper.writeValueAsString(stats);
        assertTrue(serializedStats.contains("lastFailureException"));
        assertTrue(serializedStats.contains("java.net.SocketTimeoutException"));
    }
```